src/main/java/Task.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * A class of task.
+ * Contains a description of task and a boolean to indicate the status of task.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructor.
+     * @param description a description of a task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Overloading constructor.
+     * @param description a description of a task.
+     * @param isDone a boolean show the status of task.
+     */
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Returns the description of the task.
+     * @return description of task.
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**\
+     * Returns the icon for corresponding status of task.
+     * @return sign of tick or cross.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
+    }
+
+    /**
+     * Returns immutable task with done status.
+     * @return new task with done status.
+     */
+    public Task markAsDone() {
+        return new Task(this.description, true);
+    }
+
+    /**
+     * Returns a string of task in format of task in data file.
+     * @return string of a task.
+     */
+    public String stringify() {
+        return this.toString();
+    }
```

Comment:
Is this method necessary since it is just calling the overriden `toString()` method in line 65?

Reply:
As you can see, task is a parent class of Done, Todo, Deadline and Event class. The stringify method in task class will be overridden in sub classes, hence the string returned by this method will be more specific to the corresponding object. Due to polymorphism, I will refer all these sub class objects to Task. In particular, all these tasks were gotten out by looping through the list of tasks when the method Stringify is called, this will cause runtime error because the stringify could not be found in the Task class even if it exists in child class. The function of stringify is to print the object in line for data file while the toString method is to print the object in line for list command. Therefore, this stringify method is not omissive. In order to make the function declaration be more specific, I could rename it to be toStringForDataFile. What do you think.

Reply:
Ah okay I understand what you're trying to accomplish! In that case maybe you could consider making `Task `an abstract class (if your current code allows for it), so that you may declare `stringify` to be an abstract method so you wouldn't need to give an implementation for it in `Task` (this is just a suggestion though, feel free to ignore this!). I do agree that your suggested method name is much clearer 👍🏽 Thanks for the clarification

Reply:
That is also a good idea. I will try it. Thanks!